frontend/src/js/route-config/route-config.js

```diff
@@ -518,20 +591,32 @@
         $rootScope.isAuth = false;
         // check for valid user
         $rootScope.$on('$stateChangeStart', function(event, toState) {
-            if (toState.authenticate && !utilities.isAuthenticated()) {
-                $rootScope.isAuth = false;
-                // User isn’t authenticated
-                $state.transitionTo("auth.login");
-                event.preventDefault();
-            }
-            // restrict authorized user too access login/signup page
-            else if (toState.authenticate === false && utilities.isAuthenticated()) {
-                $rootScope.isAuth = true;
-                $state.transitionTo("home");
-                event.preventDefault();
-                return false;
-            } else if (utilities.isAuthenticated()) {
+            var path = $location.path(); // get the current path
+            var pathArray = path.split('/'); // split it to make a array
+            var challengeId = pathArray[pathArray.length-2]; // get the challenge id
+
+            if (toState.url === '/leaderboard' || toState.url === '/overview'
+                || toState.url === '/evaluation' || toState.url === '/phases'
```

Comment:
Instead of using these conditions just take challenge as reference.

Reply:
I am going with these conditions for now. @RishabhJain2018 can you please incorporate @aka-jain's commit in a separate pull request?